### ISSUE TITLE
Typescript and Javascript highlighting don't work

### ISSUE THREAD

**Issue by Yappaholic** [igor-ramazanov/kak-tree-sitter-helix#34]
Every other language highlighting is working well, but there is no highlighting for Typescript and Javascript

Tree-sitter logs:

```kak
shell stderr: <<<
tree-sitter 2024-08-02T18:47:50.626020392+00:00 (kak_tree_sitter_config) [DEBUG]: loading default configuration
tree-sitter 2024-08-02T18:47:50.628035544+00:00 (kak_tree_sitter_config) [DEBUG]: loading user configuration
tree-sitter 2024-08-02T18:47:50.628054115+00:00 (kak_tree_sitter_config) [DEBUG]: loading configuration at /home/savvy/.config/kak-tree-sitter/config.toml
tree-sitter 2024-08-02T18:47:50.642956725+00:00 (kak_tree_sitter) [DEBUG]: server already running, but initiating first session 239558
tree-sitter 2024-08-02T18:47:50.643000542+00:00 (kak_tree_sitter::client) [DEBUG]: connected to KTS
>>>
```

**Comment by igor-ramazanov** [comment]
Yeah, I haven't verified them yet, but it looks like Typescript and Javascript both share somewhat special handling in Helix, because there's also TSX and JSX which treated as different languages or e.g. Vue has its own language server, etc.

I haven't spent time on it yet, but, yeah, I am aware of the issue.

**Comment by igor-ramazanov** [comment]
So, here's why it doesn't work:
1. https://github.com/helix-editor/helix/blob/101a74bf6edbbfdf9b0628a0bdbbc307ebe10ff2/helix-core/src/syntax.rs#L721-L739
2. https://github.com/helix-editor/helix/blob/101a74bf6edbbfdf9b0628a0bdbbc307ebe10ff2/runtime/queries/ecma/README.md
3. https://github.com/helix-editor/helix/blob/master/runtime/queries/javascript/highlights.scm

Helix preprocesses queries replaces any mention of `inherits: (lang name)` by replacing it with the content of the referred file.

We should do the same.

**Comment by igor-ramazanov** [comment]
There are some more unsupported languages not working for the same reason:
```
astro
common-lisp
cpp
crystal
docker-compose
dune
env
erb
git-attributes
gjs
glsl
gts
helm
javascript
jsonc
jsx
llvm-mir-yaml
mojo
msbuild
nunjucks
ocaml-interface
opencl
pkgbuild
racket
rmarkdown
sage
snakemake
starlark
templ
tfvars
tsx
typescript
wast
webc
```

**Comment by Yappaholic** [comment]
I previously thought it was due to way kak-tree-sitter works with JSX and TSX files, which the dev noted [here](https://github.com/hadronized/kak-tree-sitter/blob/master/docs/man/JSX.md), but now I understand it better, thanks.

Also yes, I noticed the same in cpp, and basically if any grammar inherits from another one it is probably broken.

Maybe we can fetch the grammars and just compile them with possible fix from the kak-tree-sitter docs?

**Comment by igor-ramazanov** [comment]
@Yappaholic Do you mean generating an actual `~/.config/kak-tree-sitter/config.toml` with proper values for fetching and compiling them locally (and patching some along the way like JSX/TSX) instead of embedding into a `nix` package?

**Comment by Yappaholic** [comment]
> @Yappaholic Do you mean generating an actual `~/.config/kak-tree-sitter/config.toml` with proper values for fetching and compiling them locally (and patching some along the way like JSX/TSX) instead of embedding into a `nix` package?

Yeah, that's the general idea in my mind. To do this we can compile kak-tree-sitter with kts and use it instead of helix grammars.

The whole patching might be done like this:
1. Fetch grammars with kts
2. Apply patches where needed
3. Compile and install grammars
4. Do other stuff

Helix has some grammars that do not exist in the default kakoune config (like `yuck`), so we can try and add them in the `config.toml`.

**Comment by igor-ramazanov** [comment]
@Yappaholic Yeah, probably, makes sense, especially for non-nix folks in the future.

Although, I also had an idea of doing something else that would allow using `nix` built binaries and libs on non `nix` machines.

By the way, I've fixed it in https://github.com/igor-ramazanov/kak-tree-sitter-helix/pull/38, a bit hacky but works.

**Comment by Yappaholic** [comment]
> By the way, I've fixed it in #38, a bit hacky but works.

Nice, I'll try to check if something can be done about missing libraries, but seems like tree-sitters `config.toml` has invalid grammar paths.

**Comment by igor-ramazanov** [comment]
@Yappaholic 

>Nice, I'll try to check if something can be done about missing libraries, but seems like tree-sitters config.toml has invalid grammar paths.

fixed in https://github.com/igor-ramazanov/kak-tree-sitter-helix/pull/40
